File: AedsII/Tps/1/Java/Palindromo.java
import java.util.Scanner;

public class Palindromo {
    public static boolean ehPalindromo(String str) {
        int i = 0;
        int j = str.length() - 1;
        boolean resp = true;
        while (i < j) {
            if (str.charAt(i) != str.charAt(j)) {
                resp = false;
                i = j;
            } else {
                i++;
                j--;
            }
        }
        return resp;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        while (sc.hasNext()) {
            String str = sc.nextLine();
            if (ehPalindromo(str)) {
                System.out.println("SIM");
            } else
                System.out.println("NAO");
        }

        sc.close();
    }
}

File: AedsII/Tps/1/C/Palindromo.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h>

int tamanho(char str[])
{
    int cont = 0;
    while (str[cont] != '\0')
    {
        cont++;
    }
    return cont;
}

bool ehPalindromo(char str[])
{
    int n = tamanho(str);
    int i = 0;
    int j = n - 1;
    bool resp = true;
    while (i < j)
    {
        if (str[i] != str[j])
        {
            resp = false;
            i = j;
        }
        i++;
        j--;
    }
    return resp;
}

int main()
{
    char s1[] = "FIM";
    char str[1000];

    scanf(" %[^\n]", str);

    while (strcmp(str, s1) != 0)
    {
        if (ehPalindromo(str))
            printf("SIM\n");
        else
            printf("NAO\n");

        scanf(" %[^\n]", str);
    }

    return 0;
}
